tools/projmgr/docs/Manual/Overview.md

```diff
@@ -0,0 +1,1317 @@
+
+Options:
+  -p, --project arg   Input cproject.yml file
+  -s, --solution arg  Input csolution.yml file
+  -f, --filter arg    Filter words
+  -o, --output arg    Output directory
+  -h, --help          Print usage
+```
+
+### Commands
+
+Print list of installed packs. The list can be filtered by words provided with the option `--filter`:
+
+```text
+list packs [--filter "<filter words>"]
+```
+
+Print list of available device names. The list can be filtered by words provided with the option `--filter`:
+
+```text
+list devices [--filter "<filter words>"]
+```
+
+Print list of available components. The list can be filtered by a selected device in the `cproject.yml` file with the option `--input` and/or by words provided with the option `--filter`:
+
+```text
+list components [--p <example.cproject.yml> --filter "<filter words>"]
+```
+todo: this does not work anymore
+
+Print list of unresolved project dependencies. The device and components selection must be provided in the `cproject.yml` file with the option `--input`. The list can be filtered by words provided with the option `--filter`:
+
+```text
+list dependencies --input <example.cproject.yml> [--filter "<filter words>"]
+```
+
+Convert cproject.yml in cprj file:
+
+```text
+convert --p <example.cproject.yml>
+```
+
+# Project Examples
+
+## Minimal Project Setup
+
+Simple applications require just one self-contained file.
+
+**Simple Project: `Sample.cproject.yml`**
+
+```yml
+project:
+  compiler: AC6                    # Use Arm Compiler 6 for this project
+  device: LPC55S69JEV98:cm33_core0 # Device name (exact name as defined in the DFP) optional with Pname for core selection
+  
+  optimize: size                   # Code optimization: emphasis code size
+  debug: on                        # Enable debug symbols
+
+  groups:                          # Define file groups of the project
+    - group: My files           
+      files:                       # Add source files
+        - file: main.c
+
+    - group: HAL
+      files:
+        - file: ./hal/driver1.c
+
+  components:                      # Add software components
+    - component: Device:Startup
+```
+
+## Software Layers
+
+Software layers collect source files and software components along with configuration files for re-use in different projects.
+The following diagram shows the various layers that are used to compose the IoT Cloud examples.
+
+![Software Layers](./images/Layer.png "Target and Build Types")
+
+The following example is a `Blinky` application that uses a `App`, `Board`, and `RTOS` layer to compose the application for a NUCELO-G474RE board.  Note, that the `device:` definition is is the `Board` layer.
+
+**Example Project: `Blinky.cproject.yml`**
+
+```yml
+project:
+  compiler: AC6
+
+  layers:
+    - layer: .\Layer\App\Blinky.clayer.yml
+    - layer: .\Layer\RTOS\RTX.clayer.yml
+    - layer: .\Layer\Board\Nucleo-G474RE.clayer.yml
+```
+
+**App Layer: `.\Layer\App\Blinky.clayer.yml`**
```

Comment:
The AppLayer is multi-target.

The AppLayer has some requirements, but these would be expressed in "resources".  If it cannot be mapped to the target (because of memory issues) an error should be reported.  

I agree that this needs more work, but we need to see it in practice.